### PR TITLE
Reshuffle shared source files

### PR DESCRIPTION
**Pull request by annelo-msft**
As pre-work to address:
- [Generating SDKs outside this repo](https://github.com/Azure/azure-sdk-for-net/issues/38143)
- [Making shared source types public](https://github.com/Azure/azure-sdk-for-net/issues/28369)
- [Solving the circular dependency issue](https://github.com/Azure/azure-sdk-for-net/issues/27337)

I'm working on an accounting of our shared source files and which libraries use them.

To make this easier, I'd like to at least temporarily move files used by only a single library into that library.  If any of these files need to stay shared, please let me know because that will be useful information in our prioritization and planning around the issues listed above.

This PR specifically moves:
- ApiVersionString to TestFramework
- ObjectNotDisposedException to Search
- HttpPipelineMessageHander to Identity
- MutableJsonDocument to Core

It also removes unused references to ContentTypeUtilities.

### CODE REVIEW

File: sdk/identity/Azure.Identity/tests/HttpPipelineMessageTest.cs [hunk 1]
```diff
@@ -12,11 +12,13 @@ namespace Azure.Core.Tests
 {
     public class HttpPipelineMessageTest
     {
+        private static ResponseClassifier _classifier = new ResponseClassifier();
```
Comment (christothes) [inline sdk/identity/Azure.Identity/tests/HttpPipelineMessageTest.cs:15]: I'm curious why this change is needed

**Review by christothes — APPROVED** [review]
identity LGTM

File: sdk/identity/Azure.Identity/tests/HttpPipelineMessageTest.cs (hunk 1 above)
Comment (annelo-msft) [reply sdk/identity/Azure.Identity/tests/HttpPipelineMessageTest.cs:15]: It was using ResponseClassifier.Shared before, which looked like it just created a new static instance.  For a test, it felt like creating a copy in the test would be ok, but let me know if you disagree.

Comment (christothes) [reply sdk/identity/Azure.Identity/tests/HttpPipelineMessageTest.cs:15]: no real concerns, though if they are functionally equivalent, I assume that the static one allocates fewer times if used in other places.

**Review by amnguye — APPROVED** [review]
LGTM - storage

File: sdk/identity/Azure.Identity/tests/HttpPipelineMessageTest.cs (hunk 1 above)
Comment (annelo-msft) [reply sdk/identity/Azure.Identity/tests/HttpPipelineMessageTest.cs:15]: I agree with your assessment.  For now, I'm not sure it's worth keeping the file in shared source just to reduce allocations in a test, though.  If we think it's an issue later, we can add that to the list of things we want to consider putting back.

File: sdk/search/Azure.Search.Documents/src/Azure.Search.Documents.csproj [hunk 2]
```diff
@@ -22,7 +22,6 @@
     <Compile Include="$(AzureCoreSharedSources)AzureEventSource.cs" LinkBase="Shared" />
     <Compile Include="$(AzureCoreSharedSources)AzureKeyCredentialPolicy.cs" LinkBase="Shared" />
     <Compile Include="$(AzureCoreSharedSources)AzureResourceProviderNamespaceAttribute.cs" LinkBase="Shared\Core" />
-    <Compile Include="$(AzureCoreSharedSources)ObjectNotDisposedException.cs" LinkBase="Shared" />
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Azure.Search.Documents.csproj:25]: Are we the only ones using this? Of course it was Search-specific before as shared source, but I wonder if we shouldn't make it public in Azure.Core. It *could* be useful for any SDK and better to move now than later.

Comment (annelo-msft) [reply sdk/search/Azure.Search.Documents/src/Azure.Search.Documents.csproj:25]: It looks like Search is the only one using it right now, so if it's ok, I'd like to move it there temporarily.  We can add it to https://github.com/Azure/azure-sdk-for-net/issues/28369 as something to make public, and if you want to own and drive taking it through arch board, I think that would be great.  Let me know if you'd prefer to handle it differently!

Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Azure.Search.Documents.csproj:25]: My apologies. I didn't realize it wasn't public already. Figured it was for callers to catch. Thank you for pointing that out.

**Review by heaths — APPROVED** [review]
LGTM to me for Search

**Review by TimothyMothra — COMMENTED** [review]
LGTM - monitor
